src/FiberTraces.cc

```diff
@@ -1253,14 +1259,20 @@ namespace pfsDRPStella = pfs::drp::stella;
         cout << "FiberTrace::calcProfileSwath: iSwath = " << iSwath << ": xArray[" << iRow << "][*] = " << xArray[ndarray::view(iRow)()] << endl;
       #endif
       for (int iCol = 0; iCol < imageSwath.getShape()[1]; ++iCol){
-        if (xArray[iRow][iCol] < (*xVecMean)[0])
-          profArraySwath[iRow][iCol] = (*yVecMeanF)[0];
-        else if (xArray[iRow][iCol] > (*xVecMean)[xVecMean->size() - 1])
-          profArraySwath[iRow][iCol] = (*yVecMeanF)[yVecMeanF->size() - 1];
-        else
-          profArraySwath[iRow][iCol] = spline(xArray[iRow][iCol]);
-        if (profArraySwath[iRow][iCol] < 0.)
-          profArraySwath[iRow][iCol] = 0.;
+        /// The spline's knots are calculated from bins in x centered at the
+        /// oversampled positions in x.
+        /// Outside the range in x on which the spline is defined, which is
+        /// [min(xRange) + overSample/2., max(xRange) - overSample/2.], so to
+        /// say in the outer (half overSample), the spline is extrapolated from
+        /// the 1st derivative at the end points.
+        tmpVal = spline(xArray[iRow][iCol]);
+        profArraySwath[iRow][iCol] = (tmpVal >= 0. ? tmpVal : 0.);
```

Comment:
Negative profile values are clipped because they are not physical.
I added another comment clarifying this.